TrackingTools/TrackAssociator/plugins/HcalDetIdAssociator.h

```diff
@@ -24,13 +24,16 @@ class HcalDetIdAssociator: public CaloDetIdAssociator{
  public:
    HcalDetIdAssociator():CaloDetIdAssociator(72, 70 ,0.087){};
 
-   HcalDetIdAssociator(const edm::ParameterSet& pSet):CaloDetIdAssociator(pSet){};
+   HcalDetIdAssociator(const edm::ParameterSet& pSet):CaloDetIdAssociator(pSet)
+    ,hcalReg_(pSet.getParameter<int> ("hcalRegion"))
+    {};
    
    virtual const char* name() const override { return "HCAL"; }
 
  protected:
-
-   virtual const unsigned int getNumberOfSubdetectors() const override { return 2;}
+    
+   int hcalReg_;
+   virtual const unsigned int getNumberOfSubdetectors(int hcalReg_) const { return hcalReg_;}
```

Comment:
Yes, that seems to be the problem. (The `override` can also be restored in that case.)